### PR TITLE
Unified Login: wpcom flow

### LINKED ISSUE #2779: Unified Login: Login with WPcom flow
Implement the login with WordPress.com flow

Screen Shot 2020-08-25 at 4 11 51 PM
Screen Shot 2020-08-25 at 4 12 08 PM

### PR DESCRIPTION
**Pull request by AmandaRiu**
Closes #2779 by implementing the logic for login with wordpress.com. This PR includes the following changes:
- Login with wpcom email + password
- Login with wpcom email + magic link
- Login with wpcom -> Google signin
- Site Picker styling revamped

Not included:
- Error screens will be handled in a separate PR.
- The store picker screen is completely styled per the designs as I have some open questions about that design and will handle the feedback in a separate PR. 

### Screenshots

Prologue screen | wpcom email screen | Store picker screen
-- | -- | --
Screenshot_1598653334|Screenshot_1598653338|Screenshot_1598653462

### To Test
- Test logging in with a wordpress.com account using the password
- Test logging in with a wordpress.com account using magic link
- Test logging in with a wordpress.com account then selecting to login with google
- Test logging in with site address still works as expected

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by AmandaRiu** [comment]
### Regarding the Store Picker Design

Screen Shot 2020-08-28 at 5 12 49 PM


@Garance91540 There are a few areas that I have some questions about (any changes can be handled in a different PR):
- The label "SELECT STORE TO CONNECT" is using the secondary color which is typically used for buttons. Is this intentional? It's also formatted using the button typography settings.
- I left the radio buttons for now because I wanted to verify how a selected row should looked as only one site can be selected at a time and typically selecting using the image is for multi-select.
- What is the "Connect another store" button supposed to do?
- Does dark mode look correct?

Light | Dark 
-- | --
Screenshot_1598653462|Screenshot_1598653453


Many thanks 😄

**Comment by Garance91540** [comment]
Hi @AmandaRiu , here are my answers:

> The label "SELECT STORE TO CONNECT" is using the secondary color which is typically used for buttons. Is this intentional? It's also formatted using the button typography settings.

I was trying to align with the WP designs which use the WP blue for the "select store to connect" prompt, but looking at it again, I think you're right it' s confusing since this is not a link. I updated the design in Figma

>I left the radio buttons for now because I wanted to verify how a selected row should looked as only one site can be selected at a time and typically selecting using the image is for multi-select.

Great question. I want to align with the WP app, but can't see how this selection behavior look right now so let's leave radio buttons for now!

> What is the "Connect another store" button supposed to do?

Basically starts the Log in with site address first flow (similar behavior as in the WP app)
https://d.pr/i/1VT0LM
https://d.pr/i/CuwWsj

> Does dark mode look correct?
Yes!

One note: as you noticed, we are using different button styles in these designs. (Body 2 Medium) instead of the uppercase styling we were using before. This change should be carried in the other part of the app where we use these buttons. (Can totally be addressed later but wanted to make sure it was clear!)

**Comment by AmandaRiu** [comment]
@Garance91540 Regarding this part:

>     What is the "Connect another store" button supposed to do?
> Basically starts the Log in with site address first flow (similar behavior as in the WP app)

I think this may be better served by adding the "Login with another account" secondary button instead like on other woo login screens. The reason I say this is because on WP, "connect another site" means _adding_ another WP site to the app, but with Woo, we only show one store at a time so clicking on "Connect another store" seems odd and since they're not really _adding_ another store to the list. WDYT?

**Comment by Garance91540** [comment]
Answered in the P2! Sounds like a plan :)

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginActivity.kt [hunk 1]
```diff
@@ -127,8 +123,12 @@ class LoginActivity : AppCompatActivity(), LoginListener, GoogleListener, Prolog
         fragmentTransaction.commitAllowingStateLoss()
     }
 
-    private fun getLoginEmailFragment(): LoginEmailFragment? {
-        val fragment = supportFragmentManager.findFragmentByTag(LoginEmailFragment.TAG)
+    private fun getLoginEmailFragment(useAltLayout: Boolean): LoginEmailFragment? {
```
Comment (nbradbury) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginActivity.kt:135]: This would benefit from a comment explaining what `useAltLayout` is.

**Comment by nbradbury** [comment]
@AmandaRiu Code looks good to me, but given that I'm been away for three months I think we should get another set of eyes on this. Good with you?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginActivity.kt (hunk 1 above)
Comment (AmandaRiu) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginActivity.kt:135]: Added in 39bc1d7b0d5dd44415e26f6a8ac4082659ce6e16 👍

**Comment by ThomazFB** [comment]
Hey @AmandaRiu, really good work here. Code implementation reviewed, tested with all scenarios, also with Dark Mode and as an additional setting also using the 1Password to fill out all fields and everything is working just fine except for one scenario.

The login with a WordPress.com account with the `Continue with Google` option is not working for me on a physical device with Android 10 installed, here's what's showing up:

Whenever I choose any of my WordPress.com linked accounts the login doesn't happen and I can see an empty view quickly going down on the screen with no error prompt (which I understand from your PR description it's not something expected to be handled right now).


I've noticed that this exception appears on the LogCat when this error occurs, maybe I'm missing any configuration or is this any known issue?

```
2020-09-03 15:27:27.768 11886-11886/com.woocommerce.android D/WordPress-MAIN: GOOGLE LOGIN: Google has returned a sign in result - canceled
2020-09-03 15:27:27.774 11886-11886/com.woocommerce.android I/WooCommerce-UTILS: 🔵 Tracked: login_social_button_failure
2020-09-03 15:27:27.774 11886-11886/com.woocommerce.android E/WordPress-NUX: Google Login Failed: result was CANCELED.
2020-09-03 15:27:27.774 11886-11886/com.woocommerce.android D/WordPress-MAIN: GOOGLE SIGNUP/LOGIN: finishing signup/login
2020-09-03 15:27:27.776 11886-11886/com.woocommerce.android D/WordPress-MAIN: GOOGLE SIGNUP/LOGIN: disconnecting google client
```

Anyway, if this is a problem from my end, I'm totally fine for us to merge this.

**Comment by AmandaRiu** [comment]
@ThomazFB You'll want to make sure you've copied the `google-services.json` from `.mobile-secrets` in order to get this to work properly. None of the actual social login logic was changed so there shouldn't be a problem with this PR. @nbradbury did you test google sign in? If so I'm good with merging this PR.

**Comment by ThomazFB** [comment]
That was it for me, @AmandaRiu. Really sorry for the confusion and thanks for the help, everything is working just fine! Really awesome work! LGTM :shipit:

I'll just leave the PR approved so you can decide if you'll wait for @nbradbury feedback for your question before merging it.

**Review by ThomazFB — APPROVED** [review]

**Comment by AmandaRiu** [comment]
@ThomazFB Since you were able to get google sign-in to work please go ahead and merge 👍
